src/tap/src/tapcell.cpp

```diff
@@ -57,17 +63,1091 @@ Tapcell::~Tapcell()
 {
 }
 
-void
-Tapcell::init(Tcl_Interp *tcl_interp,
-	   odb::dbDatabase *db)
+void Tapcell::init(odb::dbDatabase *db, utl::Logger* logger)
 {
   db_ = db;
+  logger_ = logger;
+}
+
+void Tapcell::reset()
+{
+  Tapcell::phy_idx = 0;
+  Tapcell::filled_sites.clear();
+}
+
+//}
+
+//---------------------------------------------------------------
+
+void Tapcell::clear (const char* tap_prefix, const char* endcap_prefix) {
+  int taps_removed = removeCells(tap_prefix);
+  int endcaps_removed = removeCells(endcap_prefix);
```

Comment:
what is the purpose of these variables, they aren't used.

Reply:
They are needed to store the return of removeCells function which destroys/ resets the instances previously inserted and returns the number of instances destroyed. I think I can change the function removeCells to a void if you think that is better.